src/client.ts

```diff
@@ -398,5 +411,8 @@ export function buildRequestHeaders(baseHeaders?: Headers, customUserAgent?: str
   if (customCookie) {
     returnHeaders["Cookie"] = customCookie;
   }
+  if (APIKeyPortal) {
+    returnHeaders["Authorization"] = `Bearer ${APIKeyPortal}`;
```

Comment:
The API key needs to be passed via its own header.
```suggestion
    returnHeaders["Skynet-API-Key"] = APIKeyPortal;
```

If we want to add the authorization header as a way to authenticate here (which is great idea, IMO), we can do it like this (I'm just copy-pasting the entire method from my editor because I touched several places):
```js
/**
 * Helper function that builds the request headers.
 *
 * @param [baseHeaders] - Any base headers.
 * @param [customUserAgent] - A custom user agent to set.
 * @param [customCookie] - A custom cookie.
 * @param [APIKeyPortal] - API key to use for a Skynet portal.
 * @param [customJWT] - A custom JWT to use for a Skynet portal.
 * @returns - The built headers.
 */
export function buildRequestHeaders(
  baseHeaders?: Headers,
  customUserAgent?: string,
  customCookie?: string,
  APIKeyPortal?: string,
  customJWT?: string
): Headers {
  const returnHeaders = { ...baseHeaders };
  // Set some headers from common options.
  if (customUserAgent) {
    returnHeaders["User-Agent"] = customUserAgent;
  }
  // There are three ways to authenticate a user in front of the portal.
  // Their order of precedence is:
  // API Key > Authorization header > Cookie.
  // This means that if we find an API key we won't even look for a cookie.
  if (APIKeyPortal) {
    returnHeaders["Skynet-API-Key"] = APIKeyPortal;
  }
  if (customJWT) {
    returnHeaders["Authorization"] = `Bearer ${customJWT}`;
  }
  if (customCookie) {
    returnHeaders["Cookie"] = customCookie;
  }
  return returnHeaders;
}
```